homeassistant/components/wiz_light/light.py

```diff
@@ -0,0 +1,419 @@
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = kwargs.get(ATTR_BRIGHTNESS)
+
+        colortemp = None
+        if ATTR_COLOR_TEMP in kwargs:
+            kelvin = color_utils.color_temperature_mired_to_kelvin(
+                kwargs[ATTR_COLOR_TEMP]
+            )
+            colortemp = kelvin
+            _LOGGER.debug(
+                "[wizlight %s] kelvin changed and send to bulb: %s",
+                self._light.ip,
+                colortemp,
+            )
+
+        sceneid = None
+        if ATTR_EFFECT in kwargs:
+            sceneid = self._light.get_id_from_scene_name(kwargs[ATTR_EFFECT])
+
+        if sceneid == 1000:  # rhythm
+            pilot = PilotBuilder()
+        else:
+            pilot = PilotBuilder(
+                rgb=rgb, brightness=brightness, colortemp=colortemp, scene=sceneid
+            )
+        await self._light.turn_on(pilot)
+
+    async def async_turn_off(self, **kwargs):
+        """Instruct the light to turn off."""
+        await self._light.turn_off()
+
+    @property
+    def color_temp(self):
+        """Return the CT color value in mireds."""
+        return self._temperature
+
+    @property
+    def min_mireds(self):
+        """Return the coldest color_temp that this light supports."""
+        if self._bulbtype:
+            return self.kelvin_max_map()
+        # fallback
+        return color_utils.color_temperature_kelvin_to_mired(6500)
+
+    @property
+    def max_mireds(self):
+        """Return the warmest color_temp that this light supports."""
+        if self._bulbtype:
+            return self.kelvin_min_map()
+        # fallback
+        return color_utils.color_temperature_kelvin_to_mired(2500)
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        if self._bulbtype:
+            return self.featuremap()
+        # fallback
+        return SUPPORT_BRIGHTNESS | SUPPORT_COLOR | SUPPORT_COLOR_TEMP | SUPPORT_EFFECT
+
+    @property
+    def effect(self):
+        """Return the current effect."""
+        return self._effect
+
+    @property
+    def effect_list(self):
+        """Return the list of supported effects."""
+        if self._bulbtype:
+            # Special filament bulb type
+            if self._bulbtype.name == "ESP56_SHTW3_01":
+                return [self._scenes[key] for key in [8, 9, 14, 15, 17, 28, 29, 31]]
+            # Filament bulb without white color led
+            if self._bulbtype.name == "ESP06_SHDW9_01":
+                return [self._scenes[key] for key in [8, 9, 13, 28, 30, 29, 31]]
+            # Filament bulb ST64
+            if self._bulbtype.name == "ESP06_SHDW1_01":
+                return [self._scenes[key] for key in [8, 9, 13, 28, 29, 31]]
+            if self._bulbtype.name == "ESP15_SHTW1_01I":
+                return [
+                    self._scenes[key]
+                    for key in [5, 8, 9, 10, 11, 12, 13, 14, 15, 17, 28, 30, 29, 31]
```

Comment:
Constants would be nice for these as well
```suggestion
            if self._bulbtype.name == "ESP56_SHTW3_01":
                return [self._scenes[key] for key in FILAMENT_BULB_EFFECTS]
            # Filament bulb without white color led
            if self._bulbtype.name == "ESP06_SHDW9_01":
                return [self._scenes[key] for key in FILAMENT_COLOR_BULB_EFECTS]
            # Filament bulb ST64
            if self._bulbtype.name == "ESP06_SHDW1_01":
                return [self._scenes[key] for key in ST64_BULB_EFECTS]
            if self._bulbtype.name == "ESP15_SHTW1_01I":
                return [
                    self._scenes[key]
                    for key in SHTW1_BULB_EFFECTS
```